core/storage/base_model/gae_models.py

```diff
@@ -889,17 +889,19 @@ def _require_not_marked_deleted(self) -> None:
         if self.deleted:
             raise Exception('This model instance has been deleted.')
 
-    # TODO(#13523): Change 'snapshot' to domain object/TypedDict to
-    # remove Any from type-annotation below.
+    # Here we use type Any because the method 'compute_snapshot' defined in
+    # subclasses of VersionedModel can return different Dict/TypedDict types.
+    # So, to allow every Dict/TypedDict type we used any here.
```

Comment:
any --> Any

That said, shouldn't this be 'union of typedicts'? I don't think I buy that this can't be reduced from Any.

Reply:
No we cannot use Union here because:
```
class A:
   def foo(x: Union[str, int, bool]) -> None:
         pass

class B(A):
     def foo(x: int) -> None:  # Here Mypy throw an override error because function signature does not match with base class.
         pass

class C(A):
     def foo(x: Union[str, int]) -> None:  # Here Mypy throw an override error because function signature does not match with base class.
         pass

```

Similar reason goes for return values.

Reply:
I'm not sure I buy the override error argument, because it would make sense to override if a class was more specific than its parent. However, since this is on BaseVersionedModel I think it's OK to keep it generic.

Also I think you didn't address the typo I mentioned initially.

Reply:
change `any --> Any` is added